src/main/java/dev/dfonline/codeclient/dev/overlay/ActionViewer.java

```diff
@@ -0,0 +1,184 @@
+import org.joml.Vector2i;
+import org.joml.Vector2ic;
+
+import java.util.Iterator;
+import java.util.List;
+
+public class ActionViewer {
+    private static Action action;
+    private static boolean reference = false;
+    private static ItemStack book;
+
+    public static void invalidate() {
+        action = null;
+        reference = false;
+        book = null;
+    }
+    public static void onClickChest(BlockHitResult hitResult) {
+        invalidate();
+        var world = CodeClient.MC.world;
+        if (world == null || !Config.getConfig().ActionViewer) return;
+        var position = hitResult.getBlockPos();
+        if (CodeClient.location instanceof Dev dev && dev.isInDev(position) && world.getBlockEntity(position) instanceof ChestBlockEntity) {
+            var signEntity = world.getBlockEntity(position.down().west());
+            if (signEntity instanceof SignBlockEntity sign) {
+                if (sign.getFrontText().getMessage(0, true).contains(Text.literal("CALL FUNCTION"))) {
+                    reference = true;
+                    return;
+                }
+                try {
+                    action = ActionDump.getActionDump().findAction(sign.getFrontText());
+                } catch (Exception ignored) {
+
+                }
+            }
+        }
+    }
+
+    public static ItemStack getReferenceBook() {
+        var player = CodeClient.MC.player;
+        if (player == null) return null;
+        var inventory = player.getInventory();
+
+        ItemStack referenceBook = null;
+        for (int index = 0; index < inventory.size(); index++) {
+            var itemStack = inventory.getStack(index);
+            if (itemStack.isEmpty()) continue;
+            if (itemStack.getItem() instanceof WrittenBookItem item) {
+                NbtCompound nbt = itemStack.getNbt();
+                if (nbt == null) continue;
+                var data = nbt.getCompound("PublicBukkitValues");
+                var instance = data.getString("hypercube:item_instance");
+                if (instance == null) continue;
+                referenceBook = itemStack;
+                break;
+            }
+        }
+        if (referenceBook == null || referenceBook.getName().getString().contains("◆ Reference Book ◆")) return null;
+        return referenceBook;
+    }
+
+    public static List<Text> getOverlayText() {
+        if (CodeClient.location instanceof Dev dev) {
+            if (!Config.getConfig().ActionViewer) return null;
+            ItemStack item;
+            if (action == null) {
+                if (!reference) return null;
+                if (book == null) {
+                    item = getReferenceBook();
+                    if (item == null) return null;
+                    var displayBook = item.copy();
+                    var name = displayBook.getName().copy().withColor(0xFFFFFF);
+                    displayBook.setCustomName(name); // continuity with actions
```

Comment:
Actions have no colour since you don't have an actiondump with colors:
- Running just `/getactiondump` will achieve this.
- Running `/getactiondump section` will get an actiondump with colours, which is intented behaviour.
tl;dr, they are supposed to have color